### PR TITLE
Added NL query context management. 

### PR DESCRIPTION
**Pull request by dwnoble**
* Added application state management using [easy-peasy](https://easy-peasy.vercel.app/)
* Clicking on a sample query now starts a new context
* "Recent" sidebar items switch between query contexts
* Improved scrolling behavior when running queries and switching contexts
* Refactored several components

Verified against go/dcnl-main-demo

Screencast:
nl-contexts

### CODE REVIEW

File: static/js/apps/nl_interface/query_welcome.tsx [hunk 1]
```diff
@@ -58,11 +47,13 @@ export function QueryWelcome(props: QueryWelcomePropType): JSX.Element {
...
-      setExampleQueries(
-        topicQueries[topic.current].filter((q) => q.wai).slice(0, 12)
-      );
+    const topicQueries = _.groupBy<SampleQuery>(
+      allQueries,
+      "topic"
+    ) as TopicQueries;
+    setTopicQueries(topicQueries);
+    if (topic && topicQueries[topic]) {
+      setExampleQueries(topicQueries[topic].filter((q) => q.wai).slice(0, 12));
```
Comment (chejennifer) [inline static/js/apps/nl_interface/query_welcome.tsx:56]: nit: make 12 a constant along the lines of MAX_EXAMPLE_QUERIES

File: static/js/apps/nl_interface/app_state.ts [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+  id: string;
+  name: string;
+  nlQueryIds: string[];
+}
+
+interface NLAppConfig {
+  autoRun: boolean;
+  currentNlQueryContextId: string;
+  delayDisabled: boolean;
+  demoMode: boolean; // Show feedback buttons
```
Comment (chejennifer) [inline static/js/apps/nl_interface/app_state.ts:81]: I think demoMode hides the feedback buttons right?

**Review by chejennifer — APPROVED** [review]
nice!

**Comment by pradh** [comment]
Thanks Dan.  I'll take a look with an eye towards regression in 2-3 hrs.

File: static/js/apps/nl_interface/app_state.ts (hunk 2 above)
Comment (dwnoble) [reply static/js/apps/nl_interface/app_state.ts:81]: Fixed!

File: static/js/apps/nl_interface/query_welcome.tsx (hunk 1 above)
Comment (dwnoble) [reply static/js/apps/nl_interface/query_welcome.tsx:56]: Done!

**Pull request merged**